Src/AutoFixture.NUnit3/ITypedValueProvider.cs

```diff
@@ -0,0 +1,10 @@
+﻿using System;
+
+namespace Ploeh.AutoFixture.NUnit3
+{
+    public interface ITypedValueProvider
```

Comment:
Hmm so this might be a bad name. The purpose of it is to expose capacities to provide values with different purposes - hence the methods CreateValue(Type) and CreateFrozenValue(Type).

I agree on the concern of exposure of public interfaces which may cause confusion.